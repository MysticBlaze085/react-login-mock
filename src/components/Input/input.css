.input-view {
  width: 100%;
}

.input {
  padding: 10px;
  width: 90%;
  padding-left: 15px;
  border: 1px solid #c5c3c3;
  margin-top: 20px;
  border-radius: 4px;
  outline: none;
}

.input-error {
  border: 1px solid tomato;
}

.error {
  color: tomato;
  font-size: 14px;
  margin: 0px;
}
