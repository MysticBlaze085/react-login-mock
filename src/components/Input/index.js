import React from "react";
import PropTypes from "prop-types";

import "./input.css";

function Input({ onChange, error, ...props }) {
  return (
    <div className="input-view">
      <input
        className={`input ${error && "input-error"}`}
        onChange={onChange}
        {...props}
      />
      {error && (
        <p data-testid="error-message" className="error">
          {error}
        </p>
      )}
    </div>
  );
}

Input.propTypes = {
  onChange: PropTypes.func,
  error: PropTypes.string,
};

Input.defaultProps = {
  onChange: () => {},
  error: "",
};

export default Input;
