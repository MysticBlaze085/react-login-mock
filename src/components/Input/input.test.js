import React from "react";
import { render, cleanup } from "@testing-library/react";

import Input from "./index";

test("render input component", () => {
  const { asFragment, getByRole, getByTestId, rerender } = render(
    <Input onChange={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByRole("textbox")).toBeInTheDocument();

  rerender(<Input error="Invalid" onChange={() => {}} />);
  expect(getByRole("textbox")).toHaveClass("input-error");
  expect(getByTestId("error-message")).toBeInTheDocument();
});

// test("on change event by input", () => {
//   const { getByRole } = render(<Input onChange={() => {}} />);
// });
