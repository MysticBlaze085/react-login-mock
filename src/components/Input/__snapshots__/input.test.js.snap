// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input component 1`] = `
<DocumentFragment>
  <div
    class="input-view"
  >
    <input
      class="input "
    />
  </div>
</DocumentFragment>
`;
